### PR TITLE
📨Debug unsynced data via mail

### PR DESCRIPTION
**Pull request by semanticer**
## What's this?
This PR adds one more option to our hidden debug menu that you can access via long press on **About**. This option triggers Intent that can be handled for example by Gmail or Google Drive app. It will attach a file with the currently usynced data dump so you can send it to yourself.

### Relationships

This should address the Android part of #7309

## Why do we want this?
We want to make unsynced data testing easier even for people with no knowledge of Android internals.

## How is it done?
The internal dump file is copied to a temporary file which is then sent via email or uploaded to google drive. I had to use this temporary file that is saved to external storage because Gmail's security policy prohibits attachments of files from internal storage.

### Why not another way?
We can probably make it a bit nicer, this was sort of hacked just to prove the concept. So let me know if you have any ideas. However, I don't think we should spend too much time on debug functionality.

### Side effects
none that am aware of

## Review hints
Just try to send yourself an email.

## :squid: Permissions
Anyone

### CODE REVIEW

File: Toggl.iOS.SiriExtension.UI/IntentViewController.cs [hunk 1]
```diff
@@ -5,7 +5,6 @@
 using IntentsUI;
 using System;
 using System.Globalization;
-using Toggl.iOS.Intents;
```
Comment (heytherewill) [inline Toggl.iOS.SiriExtension.UI/IntentViewController.cs]: Revert dis

**Comment by dshr** [comment]
I might be a bit out of the loop — AFAIK the hidden debug menu is not available in production, is this for internal purposes only, or will this migrate to the main settings page in time?

File: Toggl.Droid/DebugHelpers/ErrorTriggeringFragment.cs [hunk 2]
```diff
@@ -47,6 +52,7 @@ public override View OnCreateView(LayoutInflater inflater, ViewGroup container,
             var outdatedApi = createTextView("Outdated API error");
             var outdatedAppPermanently = createTextView("Permanent outdated client error");
             var outdatedApiPermanently = createTextView("Permanent outdated API error");
+            var unsyncedDataDump = createTextView("Unsynced data dump test");
```
Comment (arturs105) [inline Toggl.Droid/DebugHelpers/ErrorTriggeringFragment.cs:55]: 🤔 Maybe "Share unsynced data dump" would be more accurate.

File: Toggl.Droid/DebugHelpers/ErrorTriggeringFragment.cs [hunk 3]
```diff
@@ -98,18 +109,15 @@ private TextView createTextView(string text)
...
-            layoutParameters.BottomMargin = margin;
-
             var textView = new TextView(Context)
             {
                 Text = text,
                 LayoutParameters = layoutParameters
             };
 
+            var padding = 20.DpToPixels(Context);
+            textView.SetPadding(padding, padding, padding,padding);
```
Comment (arturs105) [inline Toggl.Droid/DebugHelpers/ErrorTriggeringFragment.cs:119]:
```suggestion
            textView.SetPadding(padding, padding, padding, padding);
```

File: Toggl.Droid/DebugHelpers/ErrorTriggeringFragment.cs [hunk 4]
```diff
@@ -176,6 +184,29 @@ private void outdatedApiPermanentlyErrorTriggered()
...
+        {
+            void copyFile(File src, File dst) {
+                FileInputStream inStream = new FileInputStream(src);
+                FileOutputStream outStream = new FileOutputStream(dst);
+                inStream.Channel.TransferTo(0, inStream.Channel.Size(), outStream.Channel);
+                inStream.Close();
+                outStream.Close();
+            }
+
+            var backingFile = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.Personal), IUnsyncedDataPersistenceService.UnsyncedDataFileName);
```
Comment (arturs105) [inline Toggl.Droid/DebugHelpers/ErrorTriggeringFragment.cs:197]: 🤔 Maybe we can move this whole `Path.Combine` to `UnsyncedDataPersistenceService.UnyncedDataFilePath`, so callers of `UnsyncedDataPersistenceService.UnsyncedDataFileName` don't have to remember that the file is in `Environment.SpecialFolder.Personal`.

**Comment by semanticer** [comment]
Good, idea. Even I did this mistake multiple times during debugging. I think it's totally fine to generate this file right before launching the sharing option. It's fast enough and straight forward.

**Pull request merged**